model/src/main/kotlin/config/DownloaderConfiguration.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.config
+
+import org.ossreviewtoolkit.model.SourceCodeOrigin
+
+data class DownloaderConfiguration(
+    /**
+     * Configuration of the considered source code origins and their priority order.
+     */
+    val sourcesOrigin: List<SourceCodeOrigin> = listOf(SourceCodeOrigin.VCS, SourceCodeOrigin.ARTIFACT)
+) {
+    init {
+        require(sourcesOrigin.isNotEmpty()) {
+            "Empty 'sourcesOrigin' found, please specify."
```

Comment:
I think we usually just state the issue, like: "'sourcesOrigin' must not be empty."

Reply:
Because it does not make sense to configure the same source twice `sourcesOrigin` could either be a set or the distinct check could happen here already, then other places do not have to verify that it does not contain duplicates:
```
require(sourcesOrigin.size == sourcesOrigin.distinct().size) { ... }
```

Reply:
For checking duplicates, also the `.getDuplicates { it }` helper function could be used. Talking about it, I'll make `it` the default now.

Reply:
Fixed and refined the error message. Will rebase once you made `it` default.

Reply:
Sorry, I looked into it, but for some reason related to generics magic it does not seem to be easily possible to make `it` the default. So please go ahead and don't wait for me.

Reply:
Got it to work with [an overload](https://github.com/oss-review-toolkit/ort/pull/3802).